lib/fluent/plugin/in_tail.rb

```diff
@@ -213,8 +219,30 @@ def expand_paths
           paths << path
         end
       }
-      excluded = @exclude_path.map { |path| path = date.strftime(path); path.include?('*') ? Dir.glob(path) : path }.flatten.uniq
-      paths - excluded
+      excluded = @exclude_path.map {|path| path = date.strftime(path); path.include?('*') ? Dir.glob(path) : path}.flatten.uniq
+      to_return = paths - excluded
+      # filter out non existing files, so in case pattern is without '*' we don't do unnecessary work
+      to_return = to_return.select {|path|
+        Pathname.new(path).exist?
+      }
+      Hash[to_return.map {|path|
+        tuple = PathInodeTuple.new(path, Fluent::FileWrapper.stat(path).ino)
+        if @follow_inodes
+          [tuple.ino, tuple]
+        else
+          [tuple.path, tuple]
+        end
+      }]
+    end
+
+    def existence_path
+      Hash[@tails.keys.map {|path_ino|
```

Comment:
With lots of files, e.g. 1000+ files, this `map` creates lots of object.
So avoiding map is better for the performance.

Of course, `expand_paths`'s map is also same.